src/USER-CONP/fix_charge_update.cpp

```diff
@@ -173,23 +189,50 @@ void FixChargeUpdate::setup(int) {
 
   // setup pots with target potentials
   std::vector<int> mpos = local_to_matrix();
+  pots = std::vector<double>(ngroup);
   double const evscale = 0.069447;  // TODO do units properly
   for (int i = 0; i < nlocal; i++) {
     for (int g = 0; g < number_groups; g++) {
       if (mask[i] & group_bits[g]) pots[mpos[i]] = group_pots[g] * evscale;
     }
   }
-  MPI_Allreduce(MPI_IN_PLACE, pots, ngroup, MPI_DOUBLE, MPI_SUM, world);
+  MPI_Allreduce(MPI_IN_PLACE, &pots.front(), ngroup, MPI_DOUBLE, MPI_SUM,
+                world);
 
   // initial elastance matrix and b vector
   vector_compute->compute_vector();
-  if (read_matrix) {
-    read_from_file();
+  std::vector<std::vector<double>> capacitance;
+  if (read_elas) {
+    elastance = read_from_file(input_file_elas);
```

Comment:
->elas_mat?

Reply:
Ich finde elastance klarer. Dass das eine Matrix ist, sieht man ja in der Definition.